src/ReservationItem.php

```diff
@@ -660,6 +675,13 @@ public static function showListSimple()
                 }
             }
 
+            // Filter locations if location was provided/submitted
+            if (isset($_POST['locations_id']) && !empty($_POST['locations_id'])) {
+                $criteria['WHERE'][] = [
+                    'glpi_locations.id' => getSonsOf('glpi_locations', (int) $_POST['locations_id']),
```

Comment:
```suggestion
                    'glpi_locations.id' => getSonsOf('glpi_locations', $_POST['locations_id']),
```

Reply:
The `getSonsOf()` function requires an *integer* for it's second parameter `$IDf` but `$_POST['locations_id']` contains a string: why removing the `(int)` cast?

Reply:
You can keep the cast.